### PR TITLE
feat(GUI): wifi settings modal for etcher pro

### PR DESCRIPTION
**Pull request by Shou**
We add a new component for configuring WiFi on Pro devices, written
in Rendition and React with styled-components. It is hidden behind an
environment variable `ETCHER_ENABLE_WIFI_DIALOG`.

Closes: https://github.com/resin-io/etcher/issues/2252
Change-Type: patch
Changelog-Entry: WiFi settings modal for Etcher Pro.

### CODE REVIEW

**Comment by Shou** [comment]
Preliminary screenshots and due to change to better match the design.

image

File: lib/gui/app/app.js [hunk 1]
```diff
@@ -345,6 +358,20 @@ app.controller('HeaderController', function (OSOpenExternalService) {
...
+   * @function
+   * @public
+   *
+   * @returns {Boolean}
+   *
+   * @example
+   * HeaderController.shouldShowWifi()
+   */
+  this.shouldShowWifi = () => {
+    return Boolean(process.env.ETCHER_ENABLE_WIFI_DIALOG)
```
Comment (zvin) [inline lib/gui/app/app.js]: `Boolean('0')` or `Boolean('false')` is true

Comment (Shou) [reply lib/gui/app/app.js]: Good call, believe that's fixed now.

File: lib/gui/app/app.js [hunk 2]
```diff
@@ -345,6 +358,21 @@ app.controller('HeaderController', function (OSOpenExternalService) {
...
+   * @public
+   *
+   * @returns {Boolean}
+   *
+   * @example
+   * HeaderController.shouldShowWifi()
+   */
+  this.shouldShowWifi = () => {
+    const parsedEnvVar = _.attempt(JSON.parse, process.env.ETCHER_ENABLE_WIFI_DIALOG)
+    return _.isError(parsedEnvVar) ? false : parsedEnvVar
```
Comment (zvin) [inline lib/gui/app/app.js]: NIT:
```javascript
return _.isError(parsedEnvVar) ? false : Boolean(parsedEnvVar)
```

Comment (Shou) [reply lib/gui/app/app.js]: Thanks! Updated again now.

**Comment by jhermsmeier** [comment]
@Shou conflicts need resolving

File: lib/gui/app/index.html [hunk 3]
```diff
@@ -11,6 +11,13 @@
   </head>
   <body>
     <header class="section-header" ng-controller="HeaderController as header">
+      <button class="button button-link"
+        ng-if="header.shouldShowWifi()"
+        ng-click="header.openWifiSettings()"
+        tabindex="4">
```
Comment (lurch) [inline lib/gui/app/index.html]: There's now two buttons with tabindex of 4? (however I don't know if that's actually a problem or not)

File: package.json [hunk 4]
```diff
@@ -77,13 +77,15 @@
...
     "redux": "3.5.2",
+    "rendition": "4.6.0",
     "request": "2.81.0",
     "resin-cli-form": "1.4.1",
     "resin-cli-visuals": "1.4.1",
     "resin-corvus": "1.0.0-beta.31",
     "roboto-fontface": "0.9.0",
     "semver": "5.1.1",
     "speedometer": "1.0.0",
+    "styled-components": "https://registry.npmjs.org/styled-components/-/styled-components-3.2.6.tgz",
```
Comment (lurch) [inline package.json]: Hmmm? Why is this a URL rather than just a version-number?

File: lib/gui/app/app.js [hunk 5]
```diff
@@ -345,6 +359,21 @@ app.controller('HeaderController', function (OSOpenExternalService) {
...
+   * @public
+   *
+   * @returns {Boolean}
+   *
+   * @example
+   * HeaderController.shouldShowWifi()
+   */
+  this.shouldShowWifi = () => {
+    const parsedEnvVar = _.attempt(JSON.parse, process.env.ETCHER_ENABLE_WIFI_DIALOG)
+    return _.isError(parsedEnvVar) ? false : Boolean(parsedEnvVar)
```
Comment (lurch) [inline lib/gui/app/app.js]: There's _many_ other places in the Etcher codebase https://github.com/resin-io/etcher/search?q=Boolean%28process.env&type=Code where a `process.env.variable` is cast either directly or implicitly to a `Boolean`... to ensure that they all get handled consistently, I wonder if it'd be worth adding a shared utility method?

File: lib/gui/app/index.html (hunk 3 above)
Comment (Shou) [reply lib/gui/app/index.html]: Since only one element's there at a time with `ng-if` it's fine.

File: package.json (hunk 4 above)
Comment (Shou) [reply package.json]: It should be resolved by the rebase.

File: lib/gui/app/app.js (hunk 5 above)
Comment (Shou) [reply lib/gui/app/app.js]: Ideally yes, but the env var setting will be partially or wholly replaced – once that's done we'll know whether we need that or not.

**Comment by jhermsmeier** [comment]
Closing, as superseded by https://github.com/resin-io/etcher/pull/2411

**Pull request closed without merging**